src/EditorFeatures/CSharpTest/UseIndexOrRangeOperator/UseIndexOperatorTests.cs

```diff
@@ -63,6 +64,7 @@ public async Task TestSimple()
         {
             await TestAsync(
 @"
+namespace System { public struct Index { } }
```

Comment:
Does the compiler already have a version of this we can use? Does the compiler also assume extra members or something?

Reply:
> Does the compiler already have a version of this we can use?

Not that i could find.

>  Does the compiler also assume extra members or something?

I think it might.  But i think this is also a fine check for the IDE side.  We do this all the time with many other features.  i.e. we just look for the presence of the type, we dn't care if it has all the members the compiler needs.  The chances of the type existing and not being appropriate are so low that we've never coded for that.

--

Feel free to merge :)